pycloudlib/oci/cloud.py

```diff
@@ -366,7 +366,15 @@ def snapshot(self, instance, clean=True, name=None):
             current_data=image_data,
             desired_state="AVAILABLE",
         )
-
-        self.created_images.append(image_data.id)
+        
+        # only add image to created_images list for cleanup if keep is False
+        if not keep:
+            self.created_images.append(image_data.id)
+        else:
+            self.preserved_images.append({
+                "id": image_data.id,
+                "name": image_data.display_name,
+            })
```

Comment:
If we want to get exotic we could use a namedtuple for this instead of a dict keyed on id and name.